### DISCUSSION TITLE
Source of background imagery selections

### DISCUSSION THREAD

**Discussion by gdabski** [Zverik/every_door#545] [Q&A] [answered]
Hello, where do you take the WMS/WMTS URLs that are made available in the _Background imagery_ menu from? One of them in my area doesn't work and I would like to check if the URL can be adjusted, but can't localize the are they defined in code.

**Comment by deevroman** [comment]
As long as the URLs are hard-wired into the code, they can't be changed. Keep an eye on these issue: 

https://github.com/Zverik/every_door/issues/423

**Reply by gdabski** [reply]
I know they must be hard-coded, I just can't seem to identify where. I'd submit a pull request if I managed to confirm that the URL is wrong and should be adjusted.

**Reply by deevroman** [reply]
https://github.com/Zverik/every_door/blob/main/lib/providers/imagery.dart

https://github.com/Zverik/every_door/blob/main/tools/add_imagery.py

**Reply by mnalis** [reply, marked as answer]
so, it comes from https://osmlab.github.io/editor-layer-index/

**Comment by mnalis** [comment]
I'm not sure, but looking at [build instructions](https://github.com/Zverik/every_door/blob/main/README.md), it seems to be located in `presets.db`  - e.g. `sqlite3 presets.db` - `.tables` - `select * from imagery`. 

What is the authoritative upstream  source for table `imagery` in `presets.db` is left as an exercise to the reader :smile:
